drivers/include/net/netdev2.h

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *               2015 Ell-i open source co-operative
+ *               2015 Freie Universität Berlin
+ *               2014 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for
+ * more details.
+ */
+
+/**
+ * @defgroup    drivers_netdev2 Low-Level Network Driver Interface
+ * @ingroup     drivers
+ * @brief       Low-level network driver interface
+ * @{
+ *
+ * This interface is supposed to be a low-level interface for network drivers.
+ *
+ * Example call flow:
+ *
+ * 1. packet arrives -> driver calls netdev2->event_callback with
+ *    event==NETDEV_EVENT_ISR (from interrupt routine)
+ * 2. event_callback wakes up upper layer thread
+ * 3. upper layer calls netdev2->driver.isr()
+ * 4. netdev2->driver.isr() calls netdev2->event_callback() with
+ *    event==NETDEV_EVENT_RX_COMPLETE
+ * 5. netdev2->event_callback() uses netdev2->driver.recv() to fetch packet
+ *
+ * @file
+ * @brief       Definitions low-level network driver interface
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ * @author      Martine Lenders <mlenders@inf.fu-berlin.de>
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef NETDEV2_H
+#define NETDEV2_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include <sys/uio.h>
+
+#include "net/netopt.h"
+
+enum {
+    NETDEV2_TYPE_UNKNOWN,
+    NETDEV2_TYPE_RAW,
+    NETDEV2_TYPE_ETHERNET,
+    NETDEV2_TYPE_802154,
+    NETDEV2_TYPE_CC110X,
+};
+
+/**
+ * @brief   Possible event types that are send from the device driver to the
+ *          upper layer
+ */
+typedef enum {
+    NETDEV2_EVENT_ISR,           /**< driver needs it's ISR handled */
+    NETDEV2_EVENT_RX_STARTED,    /**< started to receive a packet */
+    NETDEV2_EVENT_RX_COMPLETE,   /**< finished receiving a packet */
+    NETDEV2_EVENT_TX_STARTED,    /**< started to transfer a packet */
+    NETDEV2_EVENT_TX_COMPLETE,   /**< finished transferring packet */
+    /* expand this list if needed */
+} netdev2_event_t;
+
+typedef struct netdev2 netdev2_t;
```

Comment:
Doxygen is complaining about this line, but technically its already documented in l81-86. Merge anyway?